### ISSUE TITLE
login with saved creds doesnt work after xorg upgraded to 1.17.2/RHEL7.2

### ISSUE THREAD

**Issue by tastle73** [neutrinolabs/xrdp#305]
Xrdp version 0.9.0-4
OS CentOS 7.2
X.Org 1.17.2

After xorg upgraded itself last night, I can no longer login with creds saved in the RDP client. This is the same in remina, ms rdc in osx or native ms rdc.

**Comment by metalefty** [comment]
You mean "not-saved" creds work?

**Comment by tastle73** [comment]
Yes they do oddly enough


Sent via the Samsung Galaxy S® 6, an AT&T 4G LTE smartphone<div>
</div><div>
</div><div>-------- Original message --------</div><div>From: metalefty <notifications@github.com> </div><div>Date: 1/14/2016  19:40  (GMT-05:00) </div><div>To: neutrinolabs/xrdp <xrdp@noreply.github.com> </div><div>Cc: tastle73 <tom@pcc.com> </div><div>Subject: Re: [xrdp] login with saved creds doesnt work after xorg upgraded to 1.17.2/RHEL7.2 (#305) </div><div>
</div>You mean "not-saved" creds work?

---

**Comment by metalefty** [comment]
Anyway, let's look at the log. tail -f these 2 logs and try connect with saved cred.

```
tail -f /var/log/xrdp.log
tail -f /var/log/xrdp-sesman.log
```

**Comment by tastle73** [comment]
Hi,

I just did that.  Nothing is logged in either.  That is unusual.

> On Jan 14, 2016, at 8:29 PM, metalefty <notifications@github.com> wrote:
> 
> Anyway, let's look at the log. tail -f these 2 logs and try connect with saved cred.
> 
> tail -f /var/log/xrdp.log
> tail -f /var/log/xrdp-sesman.log
> —
> Reply to this email directly or view it on GitHub <https://github.com/neutrinolabs/xrdp/issues/305#issuecomment-171837824>.
> 

Thomas Astle
System Administrator

Physician's Computer Company
20 Winooski Falls Way, Suite 7
Winooski, VT 05404
+1 800.722.7708
tom@pcc.com

**Comment by metalefty** [comment]
Running xrdp and sesman in foreground will dump more verbose logs.
That may help you.

```
xrdp --nodaemon
xrdp-sesman --nodaemon
```

**Comment by OEP** [comment]
Hi, I have a similar problem which may be this issue. The issue is with the Microsoft RDP client for OSX. The behavior is saved credentials cause the RDP client to hang at a black screen. Other clients (Remmina and Windows RDP client) work fine.

I also did not have anything useful in the logs. I have run with the `--nodaemon` flag for clues.

When a connection is received:

```
[20160217-13:57:10] [INFO ] A connection received from: 198.21.233.181 port 52628
colorDepth 0xca01 (0xca00 4bpp 0xca01 8bpp)
postBeta2ColorDepth 0xca01 (0xca00 4bpp 0xca01 8bpp 0xca02 15bpp 0xca03 16bpp 0xca04 24bpp)
highColorDepth 0x0018 (0x0004 4bpp 0x0008 8bpp 0x000f 15bpp 0x0010 16 bpp 0x0018 24bpp)
supportedColorDepths 0x000f (0x0001 24bpp 0x0002 16bpp 0x0004 15bpp 0x0008 32bpp)
earlyCapabilityFlags 0x01a1 (0x0002 want32)
got client client connection type 0x00000007
xrdp_sec_process_mcs_data_CS_SECURITY:
  client supports 40 bit encryption
  client supports 128 bit encryption
  client supports 56 bit encryption
  client supports fips encryption
Processing channel data from client - The channel is off
xrdp_sec_out_mcs_data: using no security
xrdp_load_keyboard_layout:
xrdp_load_keyboard_layout: model [] variant [] layout [us]
xrdp_caps_process_order: bitmap cache v3 supported
xrdp_caps_process_pointer: client supports new(color) cursor
xrdp_process_offscreen_bmpcache: support level 0 cache size 0 MB cache entries 0
unknown in xrdp_caps_process_confirm_active 27
unknown in xrdp_caps_process_confirm_active 25
unknown in xrdp_caps_process_confirm_active 28
xrdp_caps_process_codecs: nscodec codec id 1 prop len 3
xrdp_caps_process_codecs: rfx codec id 3 prop len 49
xrdp_process_capset_rail: rail_support_level 3
xrdp_process_capset_window wnd_support_level 2 wnd_num_icon_caches 3 wnd_num_icon_cache_entries 12
yeah, up_and_running
[20160217-13:57:10] [DEBUG] xrdp_00004e32_wm_login_mode_event_00000002
xrdp:xrdp_mm [4036484788]: xrdp_mm_create: bpp 24 mcs_connection_type 7 jpeg_codec_id 0 v3_codec_id 0 rfx_codec_id 3
[20160217-13:57:10] [WARN ] local keymap file for 0x0409 found and dosen't match built in keymap, using local keymap file
xrdp_wm_login_mode_changed: login_mode is 0
in xrdp_wm_init: 
   xrdp_wm_init: file_read_section returned non-zero, requested section not found in xrdp.ini
ERROR: The requested xrdp module not found in xrdp.ini, fall
ing back to login window
out xrdp_wm_init: 
xrdp_wm_login_mode_changed: login_mode is 1
```

When disconnecting

```
xrdp_sec_recv: xrdp_mcs_recv failed
xrdp_rdp_recv: xrdp_sec_recv failed
libxrdp_process_data: xrdp_rdp_recv failed
xrdp_process_data_in: xrdp_process_loop failed
[20160217-13:57:19] [INFO ] An established connection closed to endpoint: 198.21.233.181:52628 - socket: 8
xrdp_mcs_disconnect - socket closed
[20160217-13:57:19] [DEBUG] xrdp_mm_module_cleanup
xrdp:xrdp_encoder [4036493821]: deinit_xrdp_encoder: deiniting encoder
```

and here is my xrdp.ini

```
[globals]
bitmap_cache = true
bitmap_compression = true

[xrdp1]
name = koala-test2.cs.clemson.edu
lib = libvnc.so
username = ask
password = ask
ip = 127.0.0.1
port = -1
delay_ms = 200
```

**Comment by metalefty** [comment]
Try the latest version.
